src/encoder.rs

```diff
@@ -61,7 +65,18 @@ where
             output,
             chunks_size: chunks,
             buffer: Vec::with_capacity(0),
+            flush_after_write: false
+        }
+    }
+
+    pub fn with_flush_after_write(output: W) -> Encoder<W> {
+        Encoder {
+            output,
+            chunks_size: 8192,
+            buffer: Vec::with_capacity(0),
+            flush_after_write: true
```

Comment:
Might make sense to either add a builder, or also allow `chunks` to be configured here.